### ISSUE TITLE
private server doesn't have github sync

### ISSUE THREAD

**Issue by mozesmagyar** [screeps/screeps#15]
it would be nice if we could have the github code syncing like the official version does.

**Comment by quonic** [comment]
This seems like a great mod idea. Make it not limited to github, but just git in general.

**Comment by artch** [comment]
Why would you need to have it built into server though? Just use your local git to sync your local Steam scripts folder.

The GitHub integration on the web site is supposed to work with the web version only, since it has no "local folder" support, but the Steam client has.

**Comment by mozesmagyar** [comment]
I would want it on the server so that it pulls my code instead of needing to launch the client and push it, same as the official version afaik

**Comment by artch** [comment]
It'll be pretty complicated to set up, since external GitHub hooks need to be installed and pointed to your local server from outside.

**Comment by silasary** [comment]
Assuming you have a publicly accessible server (which is true if you're sharing with friends), it really wouldn't be that hard.

In fact, I set up the hooks by duplicating the official one, in the hope that `/api/user/github-hook` still existed, even if it wasn't in the UI. (It's not, so I got a 404).

**Comment by atav32** [comment]
Yeah it sounds like you can use git hooks for this. Or if you're really lazy, setup a `git pull origin master` in your crontab
